### PR TITLE
fix parsing of response from NewCrop

### PR DESCRIPTION
**Pull request by kchapple**
Fixes #

#### Short description of what this resolves:
Something changed in PHP, or a package dependency that broke the parsing of allergy and medication responses from NewCrop.

This fixes that, and also prevents sending diags other than ICD10 to NewCrop because NewCrop only supports ICD10.

#### Changes proposed in this pull request:

@bradymiller If possible, this should also be merged into rel-701 and added to next patch.

### CODE REVIEW

File: interface/eRxSOAP.php [hunk 1]
```diff
@@ -31,6 +31,21 @@ class eRxSOAP
     private $soapSettings = array();
     private $siteId;
 
+    protected static function fixHtmlEntities($array, $xmltoarray)
```
Comment (bradymiller) [inline interface/eRxSOAP.php:34]: Won't you need to pass $array by reference here?

Comment (bradymiller) [reply interface/eRxSOAP.php:34]: also, fyi, shooting for 7.0.2 release in several weeks (rather than another 7.0.1 patch)

Comment (kchapple) [reply interface/eRxSOAP.php]: Yes, thanks!

Comment (kchapple) [reply interface/eRxSOAP.php]: Testing a JSON solution...

File: interface/eRxSOAP.php [hunk 2]
```diff
@@ -31,19 +31,11 @@ class eRxSOAP
...
-        foreach ($array as $key => $value) {
-            if (is_array($value)) {
-                self::fixHtmlEntities($value, $xmltoarray);
-            } else {
-                $array[$key] = $xmltoarray->fix_html_entities($value);      //returns proper html values
-            }
-        }
+        $encoded = json_encode($array);
+        $fixed = $xmltoarray->fix_html_entities($encoded);
+        return json_decode($fixed, true);
```
Comment (stephenwaite) [inline interface/eRxSOAP.php:38]: smaller_excited_elephant

**Review by bradymiller — APPROVED** [review]

**Comment by bradymiller** [comment]
2DV-2(13)

**Pull request merged**